pkg/sentry/fsimpl/fuse/connection.go

```diff
@@ -302,6 +328,29 @@ func (conn *Connection) NewRequest(creds *auth.Credentials, pid uint32, ino uint
 // the kernel.Task that writes the response. See FUSE_INIT for such an
 // invocation.
 func (conn *Connection) Call(t *kernel.Task, r *Request) (*Response, error) {
+	// Block requests sent before connection is initalized.
+	if !conn.IsInitialized() && r.hdr.Opcode != linux.FUSE_INIT {
+		if t == nil {
+			conn.initNonBlockingQMu.Lock()
+			// Check again to avoid race.
+			if !conn.IsInitialized() {
```

Comment:
Is there a nicer way to avoid this double checking/race thing?

```
if t == nil && !conn.IsInitialized() && r.hdr.Opcode != linux.FUSE_INIT {
...
} else if !conn.IsInitialized() && r.hdr.Opcode != linux.FUSE_INIT {
...
}

return callDo(t, r)
```

Reply:
Or consider if simply erroring out when t == nil is reasonable.

Reply:
this is due to the data race from the lock